### DISCUSSION TITLE
The reset to upstream should also be available in the branches panel

### DISCUSSION THREAD

**Discussion by danielo515** [jesseduffield/lazygit#2520] [Ideas]
Hello.
Reading more about lazygit, I was sure that I saw a mention to reseting a branch to its upstream version. However, I was unable to find it in lazygit. After checking the article again, I found that the option is only available from the files panel.
I would love to be able to reset a branch to its upstream from the branches panel. If I could do that without checking out to the branch, that will be ideal, if not, it's a valid tradeoff.

I also find confusing that the reset options are applied from the branches panel, but no matter which branch you selected alwaays affect your current branch.

**Comment by mark2185** [comment]
>  I was sure that I saw a mention to reseting a branch to its upstream version. However, I was unable to find it in lazygit. After checking the article again, I found that the option is only available from the files panel.

I seem to recall we've already had this discussion somewhere, but for the love of me I can't find it.

> I also find confusing that the reset options are applied from the branches panel, but no matter which branch you selected alwaays affect your current branch.

Well, you're resetting the currently checked out branch _to the one you've selected_. 
Otherwise there'd have to be a menu to specify both _the one you want to reset_, and the one _you want to reset it to_.

**Reply by danielo515** [reply]
I ran today again into this problem. Any specific reason why the reset to upstream is not just another option in the modal that pops up when you hit g in the branches panel?

**Reply by mark2185** [reply]
> Any specific reason why 

Yes, it's just not implemented yet. 

There was something related in a [PR](https://github.com/jesseduffield/lazygit/issues/1985) but that never got merged, sadly. If you would be willing to pick this up, that would be great!

**Reply by danielo515** [reply]
Oh, I was asking in case it was a design decision or a limitation. I would love to contribute,but golang is a language I'm not skilled on
